x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersionExtension.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml;
+
+public interface MlConfigVersionExtension {
+    /**
+     * Returns the {@link MlConfigVersion} that Elasticsearch should use.
+     * <p>
+     * This must be at least equal to the latest version found in {@link MlConfigVersion} V_* constants.
+     */
+    MlConfigVersion getCurrentMlConfigVersion();
```

Comment:
I found when actually implementing this for serverless that passing in the fallback version is important to be able to make the correct decision in the extension on which version should be used.